warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PullupUnexecutableNodesVisitor.java

```diff
@@ -245,7 +250,12 @@ public Object visit(ASTReference node, Object data) {
         QueryPropertyMarker.Instance instance = QueryPropertyMarker.findInstance(node);
         if (instance.isType(ASTDelayedPredicate.class)) {
             JexlNode source = ASTDelayedPredicate.unwrapFully(node, ASTDelayedPredicate.class);
-            JexlNodes.swap(node.jjtGetParent(), node, source);
+            source = JexlASTHelper.dereference(source);
+            // when pulling up a delayed marker that is negated, a reference expression must be persisted
```

Comment:
I am wondering if this dereference and wrap if an and/or should be a JexlASTHelper method: JexlASTHelper.wrapIfNeeded(node)

Reply:
I can extract that logic to a local method. I'm not thinking of any broader use cases for such a method outside of this visitor. If you can think of any I'm happy to move that logic to the JexlASTHelper.